backfill/2022-08-31-decoder-failure/README.md

```diff
@@ -0,0 +1,151 @@
+# Backfill missing structured pipeline family pings
+
+Context:
+
+- [tracking bug](https://mozilla-hub.atlassian.net/browse/DSRE-999)
+
+The `structured` decoder Dataflow job entered an un-drainable state on
+2022-08-31. This introduced some pipeline latency and a small amount of data loss (about 6 million
+records) for that day, since
+[we canceled the job](https://cloud.google.com/dataflow/docs/guides/stopping-a-pipeline).
+
+The missing pings were recovered from
+[`payload_bytes_raw`](https://docs.telemetry.mozilla.org/cookbooks/bigquery/querying.html?highlight=payload_bytes_raw#projects-with-bigquery-datasets),
+the dataset we maintain for reprocessing of raw data in case of decoder failure.
+
+The purpose of this backfill is to get the missing pings from the affected period
+integrated into stable tables and downstream derived tables.
+
+## Step 1: Set up backfill project
+
+Data SRE prepared terraform for staging source and output tables to project
+`moz-fx-data-backfill-10` ([cloudops-infra PR](https://github.com/mozilla-services/cloudops-infra/pull/4356))
+including the following query logic for populating the staging source:
+
+```
+WITH raw_documents AS (
+  SELECT
+      *,
+      split(uri, "/")[safe_offset(5)] AS document_id,
+  FROM `moz-fx-data-shared-prod.payload_bytes_raw.structured`
+  WHERE DATE(submission_timestamp) = "2022-08-31" # time of incident / document loss
+), live_documents AS (
+  SELECT DISTINCT document_id
+  FROM `moz-fx-data-shared-prod.payload_bytes_decoded.structured_*`
+  WHERE DATE(submission_timestamp) = "2022-08-31" # time of incident / document loss
+)
+SELECT raw_documents.* EXCEPT (document_id) FROM raw_documents
+LEFT JOIN live_documents USING (document_id)
+WHERE live_documents.document_id IS NULL
+```
```

Comment:
We'd want a similar query for payload_bytes_error

Reply:
I updated this README with some of the gotchas which includes that we missed filtering PBE. I left this one as-is since it's what we were working from.